pkg/types/shellformat/shellformat.go

```diff
@@ -7,42 +7,82 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-// ShellFormat denotes a simple formatting modifier for shell output
-type ShellFormat = string
+// EscapeSeq denotes a simple formatting modifier / escape sequence for shell output
+type EscapeSeq = string
 
 var isNoColorTerm = os.Getenv("TERM") == "dumb" || !isTerminal(os.Stdout.Fd())
 
 // Some standard modifiers
 var (
-	Reset ShellFormat = "\033[0m"
-	Bold  ShellFormat = "\033[1m"
-
-	Black  ShellFormat = "\033[30m"
-	Red    ShellFormat = "\033[31m"
-	Green  ShellFormat = "\033[32m"
-	Yellow ShellFormat = "\033[33m"
-	Blue   ShellFormat = "\033[34m"
-	Purple ShellFormat = "\033[35m"
-	Cyan   ShellFormat = "\033[36m"
-	Gray   ShellFormat = "\033[37m"
-	White  ShellFormat = "\033[97m"
+	EscapeSeqReset  EscapeSeq = "\033[0m"
+	EscapeSeqBold   EscapeSeq = "\033[1m"
+	EscapeSeqBlack  EscapeSeq = "\033[30m"
+	EscapeSeqRed    EscapeSeq = "\033[31m"
+	EscapeSeqGreen  EscapeSeq = "\033[32m"
+	EscapeSeqYellow EscapeSeq = "\033[33m"
+	EscapeSeqBlue   EscapeSeq = "\033[34m"
+	EscapeSeqPurple EscapeSeq = "\033[35m"
+	EscapeSeqCyan   EscapeSeq = "\033[36m"
+	EscapeSeqGray   EscapeSeq = "\033[37m"
+	EscapeSeqWhite  EscapeSeq = "\033[97m"
 )
 
-// FormatShell modifies the provided string using the list of modifiers
+// Format denotes an abstract format for shell output
+type Format uint64
+
+// Formatting codes suitable for logical combination via '|'
+const (
+	Bold   Format = 1 << 1
+	Black  Format = 1 << 2
+	Red    Format = 1 << 3
+	Green  Format = 1 << 4
+	Yellow Format = 1 << 5
+	Blue   Format = 1 << 6
+	Purple Format = 1 << 7
+	Cyan   Format = 1 << 8
+	Gray   Format = 1 << 9
+	White  Format = 1 << 10
+
+	maxFormat = White
+)
+
+var allFormats = []EscapeSeq{
```

Comment:
Nitpick: this could be `[maxFormat]EscapeSeq`

Reply:
Nah, not really. `maxFormat` is `1 << 10`, so 1024, that's not what you want I guess :wink: - I though about making the shift (`10`) a constant, but then what? If I need to add a new escape sequence I need to touch all of that anyway, and the unit test `TestPackageConsistency` ensure that I don't forget to adapt the slice here...